examples-draft/SimpleExampleWF_Max.tla

```diff
@@ -0,0 +1,213 @@
+--------------------------- MODULE SimpleExampleWF_Max -----------------------
+(*
+That's an example of a proof of Spec => <>P. Such proof has to be based on
+a fairness condition. The general idea is that we have to show that either
+<>P is true, or an action is enabled that will make it true later.
+
+The example here defines an operator Max, then an algorithm that
+"calculates" the maximum in several steps (MaxAsFSM) and finally proves
+that the algorithm eventually will give the same result as the
+operator Max (a kind of refinement). In this way we model a non-atomic
+implementation of an operation defined in a declarative way.
+
+In the example SimpleExampleWF_Flag the variable was a boolean, thus we
+managed to prove the P \/ ENABLED A just from the definitions of P and A.
+Here we cannot do that and induction is needed to show P \/ ENABLED A as
+an inductive invariant.
+*)
+EXTENDS Naturals, TLAPS
+
+Max(a, b) ==
+  IF a > b THEN a ELSE b 
+
+MaxProps == \A a, b \in Nat:
+              /\ Max(a, b) >= a
+              /\ Max(a, b) >= b
+              /\ Max(a, b) \in Nat
+              /\ Max(a, b) \in {a, b}
+THEOREM MaxIsGEQ == MaxProps
```

Comment:
Wouldn't it be easier to get rid of one of the names MaxProps or MaxIsGEQ and just state the theorem?